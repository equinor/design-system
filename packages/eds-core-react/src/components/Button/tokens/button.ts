import { tokens } from '@equinor/eds-tokens'
import { ButtonToken } from '../Button.types'

const {
  typography: {
    navigation: { button: buttonTypography },
  },
  colors: {
    interactive: {
      disabled__text: { rgba: disabledTextColor },
      disabled__border: { rgba: disabledBorderColor },
      disabled__fill: { rgba: disabledColor },
    },
  },
  spacings: {
    comfortable: { medium },
  },
  shape: {
    button: { minHeight: buttonHeight, borderRadius: buttonBorderRadius },
    _modes: {
      compact: {
        button: { minHeight: compactButtonHeight },
      },
    },
  },
  interactions: { focused: outline },
  clickbounds: {
    default__base: clicboundHeight,
    compact__standard: compactClickboundHeight,
  },
} = tokens

export const button: ButtonToken = {
  background: 'transparent',
  height: `var(--eds_button__height, ${buttonHeight})`,
  typography: {
    ...buttonTypography,
    textAlign: 'center',
    fontSize: `var(--eds_button__font_size, ${buttonTypography.fontSize})`,
  },
  border: {
    type: 'border',
    width: `var(--eds_button__border_width, 1px)`,
    color: 'transparent',
    radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
    style: 'solid',
  },
  spacings: {
    top: 'var(--eds_button__padding_y, 0)',
    bottom: 'var(--eds_button__padding_y, 0)',
    left: `var(--eds_button__padding_x, ${medium})`,
    right: `var(--eds_button__padding_x, ${medium})`,
  },
  clickbound: {
    height: clicboundHeight,
    width: '100%',
    offset: {
      top: `${(parseInt(clicboundHeight) - parseInt(buttonHeight)) / 2 + 1}px`,
      left: '0',
    },
  },
  entities: {
    icon: {
      height: 'var(--eds_button__icon__size, 24px)',
      width: 'var(--eds_button__icon__size, 24px)',
    },
  },
  states: {
    hover: {
      border: {
        type: 'border',
        width: '1px',
        color: 'transparent',
        radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
        style: 'solid',
      },
    },
    focus: {
      outline: {
        type: 'outline',
        offset: '2px',
        style: 'dashed',
        color: outline.color,
        width: outline.width,
      },
    },
    disabled: {
      background: disabledColor,
      border: {
        type: 'border',
        width: '1px',
        color: disabledBorderColor,
        style: 'solid',
      },
      typography: {
        ...buttonTypography,
        color: disabledTextColor,
        textAlign: 'center',
      },
    },
  },
  modes: {
    compact: {
      height: `var(--eds_button__height_compact, ${compactButtonHeight})`,
      spacings: {
        top: 'var(--eds_button__padding_y_compact, 0)',
        bottom: 'var(--eds_button__padding_y_compact, 0)',
      },
      clickbound: {
        height: compactClickboundHeight,
        width: '100%',
        offset: {
          top: `${
            (parseInt(compactClickboundHeight) -
              parseInt(compactButtonHeight)) /
              2 +
            1
          }px`,
          left: '0',
        },
      },
    },
  },
}
