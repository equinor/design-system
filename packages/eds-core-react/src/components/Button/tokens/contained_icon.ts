import { tokens } from '@equinor/eds-tokens'
import { mergeDeepRight } from 'ramda'

import {
  primary as primaryContained,
  secondary as secondaryContained,
  danger as dangerContained,
} from './contained'
import { ButtonToken } from '../Button.types'

const {
  clickbounds: {
    default__base: clicboundHeight,
    compact__standard: compactClickboundHeight,
  },
  shape,
} = tokens

const contained_icon = {
  height: shape.icon_button.minHeight,
  width: shape.icon_button.minWidth,
  border: {
    width: '0px',
    radius: '50%',
  },
  spacings: { left: '0', right: '0' },
  clickbound: {
    width: clicboundHeight,
    offset: {
      top: `${
        (parseInt(clicboundHeight) - parseInt(shape.icon_button.minWidth)) / 2
      }px`,
      left: `${
        (parseInt(clicboundHeight) - parseInt(shape.icon_button.minWidth)) / 2
      }px`,
    },
  },
  states: {
    hover: {
      border: {
        width: '0px',
        radius: '50%',
      },
    },
  },
  modes: {
    compact: {
      height: shape._modes.compact.icon_button.minHeight,
      width: shape._modes.compact.icon_button.minWidth,
      clickbound: {
        width: compactClickboundHeight,
        offset: {
          top: '0',
          left: `${
            (parseInt(compactClickboundHeight) -
              parseInt(shape._modes.compact.icon_button.minWidth)) /
            2
          }px`,
        },
      },
    },
  },
}

export const primary = mergeDeepRight(
  primaryContained,
  contained_icon,
) as ButtonToken

export const secondary = mergeDeepRight(
  secondaryContained,
  contained_icon,
) as Partial<ButtonToken>
export const danger = mergeDeepRight(
  dangerContained,
  contained_icon,
) as Partial<ButtonToken>
