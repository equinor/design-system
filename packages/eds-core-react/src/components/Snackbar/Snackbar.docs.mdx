import { Links, PropsTable, Story } from './../../../.storybook/components'
import { Snackbar } from '.'

# Snackbar

Provides brief, temporary, non-interrupting notifications at the bottom of the screen.

<Links 
    ariaUrl="https://www.w3.org/TR/wai-aria-1.1/#alert"
    figmaUrl="https://www.figma.com/file/0bGXR2sCwMVSDNyyzu5BXrO5/UI%E2%80%94User-Interface?node-id=17659%3A40216"
    documentationUrl="https://eds.equinor.com/0b0c666ab/p/626020-snackbar/b/15cfd0"
    sourceUrl="https://github.com/equinor/design-system/blob/develop/packages/eds-core-react/src/components/Snackbar/Snackbar.tsx"
    npmUrl="https://www.npmjs.com/package/@equinor/eds-core-react"
/>

<Story id="feedback-snackbar--introduction"/>
<PropsTable />

## Usage

`Snackbars` inform users of an activity that an app has performed or will perform. <br />
 <ul>
    <li>Only one snackbar may be displayed at a time.</li>
    <li>Snackbars appear temporarily, and disappear on their own without claiming user input to be dismissed.</li>
    <li>Snackbars are placed in the most suitable area of a UI, mainly at the bottom of the UI and in a front of app content.</li>
    <li>Avoid placing a snackbar in front of frequently used touch targets or navigation.</li>
  </ul>

  ```tsx
import { Snackbar } from '@equinor/eds-core-react'


<Snackbar>Snackbar text</Snackbar>
```

## Examples

### Simple

<ul>
    <li>Snackbars appear without warning, and don not require user interaction.</li>
    <li>When multiple snackbar updates are necessary, they should appear one at a time.</li>
</ul>

<Story id="feedback-snackbar--simple"/>

### With action

A snackbar can contain a single action, a single text button that lets users take action on a process performed by the app.

<Story id="feedback-snackbar--with-action"/>

### Compact

Compact `Snackbar` using `EdsProvider`

<Story id="feedback-snackbar--compact"/>


