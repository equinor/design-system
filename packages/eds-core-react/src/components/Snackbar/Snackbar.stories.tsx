import { useState } from 'react'
import { Snackbar, SnackbarProps, Button } from '../..'
import { StoryFn, Meta } from '@storybook/react'
import { Stack } from './../../../.storybook/components'
import page from './Snackbar.docs.mdx'

const meta: Meta<typeof Snackbar> = {
  title: 'Feedback/Snackbar',
  component: Snackbar,
  subcomponents: { Action: Snackbar.Action },
  parameters: {
    docs: {
      page,
      source: {
        excludeDecorators: true,
      },
    },
  },
  decorators: [
    (Story) => {
      return (
        <Stack>
          <Story />
        </Stack>
      )
    },
  ],
}

export default meta

export const Introduction: StoryFn<SnackbarProps> = (args) => {
  const { open } = args
  const [visible, setVisible] = useState(open)
  return (
    <>
      <Button type="button" onClick={() => setVisible(true)}>
        Show a simple snackbar with default options
      </Button>
      <Snackbar {...args} open={visible} onClose={() => setVisible(false)}>
        Play with me
      </Snackbar>
    </>
  )
}

export const Simple: StoryFn<SnackbarProps> = () => {
  const [open, setOpen] = useState(false)

  return (
    <>
      <Button type="button" onClick={() => setOpen(true)}>
        Show a simple snackbar for 5 seconds
      </Button>
      <Snackbar
        open={open}
        onClose={() => setOpen(false)}
        autoHideDuration={5000}
      >
        Message goes here
      </Snackbar>
    </>
  )
}

export const WithAction: StoryFn<SnackbarProps> = () => {
  const [withActionOpen, setWithActionOpen] = useState(false)
  return (
    <>
      <Button type="button" onClick={() => setWithActionOpen(true)}>
        Show a snackbar with action for the default 7 seconds
      </Button>
      <Snackbar open={withActionOpen} onClose={() => setWithActionOpen(false)}>
        Your changes was saved
        <Snackbar.Action>
          <Button variant="ghost">Undo</Button>
        </Snackbar.Action>
      </Snackbar>
    </>
  )
}
WithAction.storyName = 'With action'

export const TimerRegressionTest: StoryFn<SnackbarProps> = () => {
  const [message, setMessage] = useState({ open: false, text: '' })
  const [data, setData] = useState(0)

  const handleClick = () => {
    setMessage({ open: !message.open, text: 'Hello, World!' })
  }

  const handleDataClick = () => {
    setData(data + 1)
  }

  return (
    <>
      <div>
        <Button onClick={() => handleClick()}>Show Snackbar</Button>
        <Button
          style={{ marginLeft: '10px' }}
          onClick={() => handleDataClick()}
        >
          Data + {data}
        </Button>
      </div>

      <Snackbar
        open={message.open}
        onClose={() => setMessage({ open: false, text: '' })}
        autoHideDuration={2000}
      >
        {message.text}
      </Snackbar>
    </>
  )
}
TimerRegressionTest.storyName =
  'Test to ensure timer stays consistent across rerenders'
//hide this story from sidebar
TimerRegressionTest.tags = ['!dev']
