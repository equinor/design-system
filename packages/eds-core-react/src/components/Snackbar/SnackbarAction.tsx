import { Children, ReactNode, forwardRef } from 'react'
import styled, { css } from 'styled-components'

const StyledSnackbarAction = styled.div(({ theme }) => {
  return css`
    display: inline-flex;
    margin-left: ${theme.entities.actions.spacings.left};
    margin-top: ${theme.entities.actions.spacings.top};
    margin-bottom: ${theme.entities.actions.spacings.bottom};
  `
})

export type SnackbarActionProps = {
  children: ReactNode
}

export const SnackbarAction = forwardRef<HTMLDivElement, SnackbarActionProps>(
  function SnackbarAction({ children, ...rest }, ref) {
    const props = {
      ...rest,
      ref,
    }
    return (
      <StyledSnackbarAction {...props}>
        {Children.only(children)}
      </StyledSnackbarAction>
    )
  },
)
