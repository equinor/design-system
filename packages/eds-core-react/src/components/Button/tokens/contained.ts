import { tokens } from '@equinor/eds-tokens'
import { ButtonToken } from '../Button.types'
import { button } from './button'

import { mergeDeepRight } from 'ramda'

const {
  colors: {
    text: {
      static_icons__primary_white: { rgba: primaryWhite },
    },
    interactive: {
      primary__resting: { rgba: primaryColor },
      primary__hover: { rgba: primaryHoverColor },
      secondary__resting: { rgba: secondaryColor },
      secondary__link_hover: { rgba: secondaryHoverColor },
      danger__resting: { rgba: dangerColor },
      danger__hover: { rgba: dangerHoverColor },
    },
  },

  shape: {
    button: { borderRadius: buttonBorderRadius },
  },
} = tokens

export const primary = mergeDeepRight(button, {
  background: primaryColor,
  typography: {
    color: primaryWhite,
  },
  border: {
    type: 'border',
    style: 'solid',
    width: '1px',
    color: primaryColor,
    radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
  },
  states: {
    hover: {
      background: primaryHoverColor,
    },
  },
}) as ButtonToken
export const secondary = mergeDeepRight(primary, {
  background: secondaryColor,
  border: {
    color: secondaryColor,
  },
  states: {
    hover: {
      background: secondaryHoverColor,
      border: {
        color: secondaryHoverColor,
      },
    },
  },
}) as ButtonToken

export const danger = mergeDeepRight(primary, {
  background: dangerColor,
  border: {
    color: dangerColor,
  },
  states: {
    hover: {
      background: dangerHoverColor,
      border: {
        color: dangerHoverColor,
      },
    },
  },
}) as ButtonToken
