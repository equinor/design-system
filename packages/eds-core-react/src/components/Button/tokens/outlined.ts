import { tokens } from '@equinor/eds-tokens'
import { mergeDeepRight } from 'ramda'
import { button } from './button'
import { ButtonToken } from '../Button.types'

const {
  colors: {
    interactive: {
      primary__resting: { rgba: primaryColor },
      primary__hover: { rgba: primaryHoverColor },
      primary__hover_alt: { rgba: primaryHoverAltColor },
      secondary__resting: { rgba: secondaryColor },
      secondary__link_hover: { rgba: secondaryHoverColor },
      secondary__highlight: { rgba: secondaryHoverAltColor },
      danger__resting: { rgba: dangerColor },
      danger__hover: { rgba: dangerHoverColor },
      danger__highlight: { rgba: dangerHoverAltColor },
    },
  },

  shape: {
    button: { borderRadius: buttonBorderRadius },
  },
} = tokens

export const primary = mergeDeepRight(button, {
  typography: {
    color: primaryColor,
  },
  border: {
    type: 'border',
    style: 'solid',
    width: '1px',
    color: primaryColor,
    radius: `var(--eds_button__radius, ${buttonBorderRadius})`,
  },
  states: {
    hover: {
      typography: {
        color: primaryHoverColor,
      },
      background: primaryHoverAltColor,
      border: {
        style: 'solid',
        type: 'border',
        width: '1px',
        color: primaryHoverColor,
      },
    },
    disabled: {
      background: 'transparent',
    },
  },
}) as ButtonToken

export const secondary = mergeDeepRight(primary, {
  typography: {
    color: secondaryColor,
  },
  border: {
    color: secondaryColor,
  },
  states: {
    hover: {
      background: secondaryHoverAltColor,
      typography: {
        color: secondaryHoverColor,
      },
      border: {
        color: secondaryHoverColor,
      },
    },
  },
}) as ButtonToken

export const danger = mergeDeepRight(primary, {
  typography: {
    color: dangerColor,
  },
  border: {
    color: dangerColor,
  },
  states: {
    hover: {
      background: dangerHoverAltColor,
      typography: {
        color: dangerHoverColor,
      },
      border: {
        color: dangerHoverColor,
      },
    },
  },
}) as ButtonToken
