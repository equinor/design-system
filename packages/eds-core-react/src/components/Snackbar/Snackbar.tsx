import { useEffect, HTMLAttributes, forwardRef, useRef, useState } from 'react'
import styled, { css, ThemeProvider } from 'styled-components'
import { snackbar as SnackbarToken } from './Snackbar.tokens'
import {
  useToken,
  typographyTemplate,
  spacingsTemplate,
  bordersTemplate,
} from '@equinor/eds-utils'
import { Paper } from '../Paper'
import { useEds } from '../EdsProvider'

type StyledProps = {
  popover: string
  $placement?:
    | 'left'
    | 'right'
    | 'top'
    | 'bottom'
    | 'top-left'
    | 'bottom-left'
    | 'top-right'
    | 'bottom-right'
}

const PopoverDiv = styled('div').withConfig({
  shouldForwardProp: () => true, //workaround to avoid warning until popover gets added to react types
})<StyledProps>(({ theme, $placement }) => {
  return css`
    inset: unset;
    border: 0;
    overflow: visible;
    position: fixed;
    padding: 0;
    background-color: transparent;
    ${{
      top: $placement.includes('top')
        ? theme.spacings.top
        : $placement === 'left' || $placement === 'right'
          ? '50%'
          : undefined,
      bottom: $placement.includes('bottom') ? theme.spacings.bottom : undefined,
      right: $placement.includes('right') ? theme.spacings.right : undefined,
      left: $placement.includes('left')
        ? theme.spacings.left
        : $placement === 'top' || $placement === 'bottom'
          ? '50%'
          : undefined,
      transform:
        $placement === 'left' || $placement === 'right'
          ? 'translateY(-50%)'
          : $placement === 'top' || $placement === 'bottom'
            ? 'translateX(-50%)'
            : undefined,
    }}
    &::backdrop {
      background-color: transparent;
    }
  `
})

const StyledSnackbar = styled(Paper)(({ theme }) => {
  return css`
    background-color: ${theme.background};
    ${spacingsTemplate(theme.spacings)}
    ${bordersTemplate(theme.border)}
    ${typographyTemplate(theme.typography)}
    min-height: ${theme.minHeight};
    box-sizing: border-box;
    a,
    button {
      color: ${theme.entities.button.typography.color};
    }
  `
})

export type SnackbarProps = {
  /**  Controls the visibility of the snackbar */
  open?: boolean
  /** How long will the message be visible in milliseconds */
  autoHideDuration?: number
  /** Callback fired when the snackbar is closed by auto hide duration timeout */
  onClose?: () => void
  /** Placement of the snackbar relative to the viewport */
  placement?:
    | 'left'
    | 'right'
    | 'top'
    | 'bottom'
    | 'top-left'
    | 'bottom-left'
    | 'top-right'
    | 'bottom-right'
} & HTMLAttributes<HTMLDivElement>

export const Snackbar = forwardRef<HTMLDivElement, SnackbarProps>(
  function Snackbar(
    {
      open = false,
      autoHideDuration = 7000,
      onClose,
      placement = 'bottom',
      children,
      ...rest
    },
    ref,
  ) {
    const [visible, setVisible] = useState(open)
    const timer = useRef<ReturnType<typeof setTimeout>>()

    useEffect(() => {
      setVisible(open)
      if (open) {
        timer.current = setTimeout(() => {
          setVisible(false)
          if (onClose) {
            onClose()
          }
        }, autoHideDuration)
      }
      return () => clearTimeout(timer.current)
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [open, autoHideDuration])

    const props = {
      ref,
      ...rest,
    }
    const { density } = useEds()
    const token = useToken({ density }, SnackbarToken)

    return (
      <ThemeProvider theme={token}>
        {visible && (
          <PopoverDiv
            popover="manual"
            $placement={placement}
            ref={(el) => el?.showPopover()}
          >
            <StyledSnackbar role="alert" elevation="overlay" {...props}>
              {children}
            </StyledSnackbar>
          </PopoverDiv>
        )}
      </ThemeProvider>
    )
  },
)
