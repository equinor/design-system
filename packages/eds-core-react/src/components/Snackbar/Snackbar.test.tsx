/* eslint-disable no-undef */
import {
  render,
  cleanup,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { Snackbar } from '.'
import { snackbar as tokens } from './Snackbar.tokens'

afterEach(cleanup)

const StyledSnackbar = styled(Snackbar)`
  clip-path: unset;
`
const message = "Hi, I'm the snackbar"

describe('Snackbar', () => {
  it('Matches snapshot', () => {
    render(<Snackbar open>snacks</Snackbar>)
    const snackbar = screen.getByText('snacks')

    expect(snackbar).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(<Snackbar open>snacks</Snackbar>)
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Can extend the css for the component', () => {
    render(
      <StyledSnackbar open data-testid="test-snackbar">
        {message}
      </StyledSnackbar>,
    )
    expect(screen.getByTestId('test-snackbar')).toHaveStyleRule(
      'clip-path',
      'unset',
    )
  })
  it('Sets the message', () => {
    render(<Snackbar open>{message}</Snackbar>)
    const snackbar = screen.getByText(message)
    expect(snackbar).toBeDefined()
  })
  it('Is only visible when the open prop is true', () => {
    const { rerender } = render(
      <StyledSnackbar open={false} data-testid="test-snackbar-1">
        {message}
      </StyledSnackbar>,
    )
    expect(screen.queryAllByTestId('test-snackbar-1')).toHaveLength(0)
    rerender(
      <StyledSnackbar open data-testid="test-snackbar-2">
        {message}
      </StyledSnackbar>,
    )
    expect(screen.getByTestId('test-snackbar-2')).toBeDefined()
  })
  it('Disappears automatically after a provided timeout', async () => {
    render(
      <Snackbar open autoHideDuration={1000}>
        {message}
      </Snackbar>,
    )
    expect(screen.getByText(message)).toBeDefined()
    await waitForElementToBeRemoved(() => screen.queryByText(message))
  })
  it('Can have one button attached', () => {
    const buttonText = "I'm the button"
    render(
      <Snackbar open>
        {message}
        <Snackbar.Action>
          <button type="button">{buttonText}</button>
        </Snackbar.Action>
      </Snackbar>,
    )
    expect(screen.getByText(buttonText)).toBeDefined()
  })
  it('Placement prop is placing the snackbar as expected', () => {
    render(
      <Snackbar open placement="top" data-testid="test-snackbar-3">
        {message}
      </Snackbar>,
    )
    expect(screen.getByTestId('test-snackbar-3')).toHaveStyleRule(
      'top',
      tokens.spacings.top,
    )
    expect(screen.getByTestId('test-snackbar-3')).toHaveStyleRule('left', '50%')
  })
})
