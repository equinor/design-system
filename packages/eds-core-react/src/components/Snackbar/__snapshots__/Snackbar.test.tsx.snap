// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Matches snapshot 1`] = `
.c0 {
  background: var(--eds_ui_background__default, rgba(255, 255, 255, 1));
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.2),0 4px 5px rgba(0, 0, 0, 0.12),0 2px 4px rgba(0, 0, 0, 0.14);
}

.c1 {
  position: fixed;
  background-color: var(--eds_ui_background__overlay, rgba(0, 0, 0, 0.8));
  padding-left: 16px;
  padding-top: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
  border-radius: 4px;
  margin: 0;
  color: var(--eds_text__static_icons__primary_white, rgba(255, 255, 255, 1));
  font-family: Equinor;
  font-size: 0.750rem;
  font-weight: 500;
  line-height: 1.333em;
  text-align: left;
  min-height: 48px;
  box-sizing: border-box;
  z-index: 1400;
  bottom: 16px;
  left: 50%;
  transform: translateX(-50%);
}

.c1 a,
.c1 button {
  color: var(--eds_interactive__link_in_snackbars, rgba(151, 202, 206, 1));
}

<div
  class="c0 c1"
  role="alert"
>
  snacks
</div>
`;
